api/pkg/block/service.go

```diff
@@ -43,6 +47,163 @@ func NewAPIService(c client.Client) *APIService {
 	}
 }
 
+func UpdateFilter(reqFilter map[string]string, filter map[string]string) error {
+	log.Debugf("Getting the filters updated for the request.")
+
+	for k, v := range filter {
+		log.Debugf("Filter Key = [%+v], Key = [%+v]\n", k, v)
+		reqFilter[k] = v
+	}
+	return nil
+}
+
+func WriteError(response *restful.Response, msg string, errCode int, err error) {
+	log.Errorf(msg, err)
+	error := response.WriteError(errCode, err)
+	if error != nil {
+		log.Errorf("Response write status and http error failed: %v\n", err)
+		return
+	}
+}
+
+func (s *APIService) checkBackendExists(ctx context.Context, request *restful.Request,
+	response *restful.Response, backendId string) error {
+
+	backendResp, err := s.backendClient.GetBackend(ctx, &backend.GetBackendRequest{Id: backendId})
+	if err != nil {
+		WriteError(response, "Get Backend details failed: %v\n", http.StatusNotFound, err)
+		return err
+	}
+	log.Infof("Backend response = [%+v]\n", backendResp)
+
+	return nil
+}
+
+func (s *APIService) prepareVolumeRequest(request *restful.Request,
+	response *restful.Response) *block.ListVolumeRequest {
+
+	limit, offset, err := common.GetPaginationParam(request)
+	if err != nil {
+		WriteError(response, "Get pagination parameters failed: %v\n", http.StatusInternalServerError, err)
+		return nil
+	}
+
+	sortKeys, sortDirs, err := common.GetSortParam(request)
+	if err != nil {
+		WriteError(response, "Get sort parameters failed: %v\n", http.StatusInternalServerError, err)
+		return nil
+	}
+
+	filterOpts := []string{"name", "type", "status"}
+	filter, err := common.GetFilter(request, filterOpts)
+	if err != nil {
+		WriteError(response, "Get filter failed: %v\n", http.StatusInternalServerError, err)
+		return nil
+	}
+
+	return &block.ListVolumeRequest{
+		Limit:    limit,
+		Offset:   offset,
+		SortKeys: sortKeys,
+		SortDirs: sortDirs,
+		Filter:   filter,
+	}
+}
+
+func (s *APIService) ListVolumes(request *restful.Request, response *restful.Response) {
+	if !policy.Authorize(request, response, "volume:list") {
```

Comment:
log before returning

Reply:
Logging has been taken care in the calling function.